public/src/components/App.jsx

```diff
@@ -41,29 +41,60 @@ function App() {
   const[siteData, setSiteData] = useState(null);
   const[checkIn, setCheckIn] = useState(null);
   const[checkOut, setCheckOut] = useState(null);
+  const[checkInSelect, setCheckInSelect] = useState(false);
+  const[checkOutSelect, setCheckOutSelect] = useState(false);
   const[showCalendar, setShowCalendar] = useState(false);
 
   function updateCheckInOut(date, checkIn, checkOut) {
-    debugger;
-    const dateString = date.toString().split(' ').slice(1, 3).join(' ');
-    if(!checkIn) {
+    const dateString = date.toString().split(' ').slice(0, 4).join(' ');
+    if(!checkIn || checkInSelect || date < new Date(checkIn.toString())) {
       setCheckIn(dateString);
-    } else if (checkIn && !checkOut) {
+      setCheckOut(null);
+      setCheckOutSelect(true);
+      setCheckInSelect(false);
+    } else if (checkIn) {
       setCheckOut(dateString);
+      setCheckOutSelect(false);
       showTotals();
     }
   }
 
   function showTotals() {
     //close calendar
     //show button
-    toggleCalendar();
+    setShowCalendar(false);
     //update price to average per night
     //if savings, show savings
     //show subtotal
   }
-  function toggleCalendar() {
-    setShowCalendar(!showCalendar)
+
+  function initialButtonClick() {
+    setShowCalendar(true)
+    setCheckInSelect(true)
+  }
+
+  function selectCheckIn() {
+    if(!showCalendar) {
+      setShowCalendar(true);
+    }
+    if (!checkInSelect) {
+      setCheckInSelect(true);
+      setCheckOutSelect(false);
+    }
+  }
+  console.log("selected checkin", checkInSelect)
+  console.log("selected checkout", checkOutSelect)
+
+  function selectCheckOut() {
+    if(!showCalendar) {
+      setShowCalendar(true);
+    }
+    if (!checkOutSelect && checkIn) {
+      setCheckOutSelect(true);
+      setCheckInSelect(false);
+    } else {
+      setCheckInSelect(true);
+    }
   }
 
   useEffect(() => {
```

Comment:
this empty useEffect function will be filled in or deleted later?

Reply:
Ah-- it just cut off after line 100 for some reason instead of line 98? The useEffect continues in the actual code to line 109. Thank you for looking out!